core/services/relay/evm/chain_reader.go

```diff
@@ -176,6 +185,10 @@ func (cr *chainReader) HealthReport() map[string]error {
 }
 
 func (cr *chainReader) GetLatestValue(ctx context.Context, contractName, method string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error {
+	if cr.StateMachine.Ready() != nil {
+		return errServiceNotStarted
+	}
```

Comment:
This presumes that the non-ready error is "not started" but that is not always the case. `StateMachine` has helpers that can be used instead. Something like:
```suggestion
	return cr.StateMachine.IfStarted(func() error {
		...
```

Reply:
Cleaner! Changes here: [`2eb9b0d` (#14123)](https://github.com/smartcontractkit/chainlink/pull/14123/commits/2eb9b0db5364574716b8f9e7d5634c592fc3b149)

Reply:
What do you think about printing the `cr.StateMachine.State()` in the new changes when `cr.StateMachine.IfStarted` returns `!ok` ? 

Could it flip to `Started` between the exec and the logging becoming an inconsistent message?